### PR TITLE
Global data new

### PR DESCRIPTION
**Pull request by Christian-B**
Part of https://github.com/SpiNNakerManchester/SpiNNUtils/pull/172

Simplified summary.

Most of the data has been moved from AbstractSimulatorBase to FecDataView

This means there is a lot less passing around of parameters and no need for injections.

Specifically for PdP2
the signature of generate_machine_data_specification has been simplified

Extra data needed can now be obtained directly on demand

This includes
FecDataView.get_routing_infos()
FecDataView.get_placements()
and a semantic sugar on placements
FecDataView.get_placement_of_vertex(vertex)

### CODE REVIEW

**Review by lplana — CHANGES_REQUESTED** [review]

**Comment by Christian-B** [comment]
Thanks you!

Very good catch!
________________________________

File: spinn_pdp2/mlp_network.py [hunk 1]
```diff
@@ -666,7 +666,7 @@ def write_Lens_output_file (self,
                         gtv = g.t_vertex[s]
                         try:
                             rec_outputs[g.write_blk].append (gtv.read (
-                                gfe.placements().get_placement_of_vertex (gtv),
+                                FecDataView.get_placement_of_vertex(ftv),
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_pdp2/mlp_network.py]: Agreed. fixed. Thanks

File: spinn_pdp2/mlp_network.py [hunk 2]
```diff
@@ -801,7 +801,7 @@ def show_test_results (self):
             ltv = g.t_vertex[g.subgroups - 1]
             try:
                 rec_test_results = ltv.read (
-                    gfe.placements().get_placement_of_vertex (ltv),
+                    FecDataView.get_placement_of_vertex(ftv),
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_pdp2/mlp_network.py]: Agreed fixed. Thanks

**Review by andrewgait — APPROVED** [review]
This looks fine to me.

**Pull request merged**